rasgotransforms/rasgotransforms/transforms/datespine/bigquery/datespine.sql

```diff
@@ -0,0 +1,40 @@
+{% set min_max_query %}
+select min(cast({{ date_col }} as date)) min_date, max(cast({{ date_col }} as date)) max_date from {{ source_table }}
+{% endset %}
+{% set min_max_query_result = run_query(min_max_query) %}
+{% if start_timestamp is defined %}
+    {% set min_date = start_timestamp %}
+{% else %}
+    {% set min_date = min_max_query_result[min_max_query_result.columns[0]][0] %}
+{% endif %}
+{% if  end_timestamp is defined %}
+    {% set max_date = end_timestamp %}
+{% else %}
+    {% set max_date = min_max_query_result[min_max_query_result.columns[1]][0] %}
+{% endif %}
+{% set row_count_query %}
+select datediff(cast('{{ min_date }}' as date), cast('{{ max_date }}' as date), {{ interval_type }})
```

Comment:
Isn't this built-in function `DATE_DIFF` in (BigQuery)[https://cloud.google.com/bigquery/docs/reference/standard-sql/date_functions] ?